### PR TITLE
also build sqlite3 executable

### PR DESCRIPTION
**Pull request by krahenbuhl**
sqlite3 is a dependency of PROJ6.x which needs both the lib and the executable to be compiled.
PROJ6 is a dependency of GDAL 3.0

### CODE REVIEW

File: conanfile_base.py [hunk 1]
```diff
@@ -7,7 +7,10 @@
 
 class ConanFileBase(ConanFile):
     _base_name = "sqlite3"
-    version = "3.29.0"
+    lib_version = "3.29.0"
+    rev = "1"
+    version = lib_version + "-" + rev
```
Comment (david-sinuela-pix4d-dev) [inline conanfile_base.py]: Well spotted

File: conanfile.py [hunk 2]
```diff
@@ -76,6 +77,10 @@ def package(self):
         self.copy("FindSQLite3.cmake")
 
     def package_info(self):
+        bin_path = os.path.join(self.package_folder, "bin")
+        self.output.info('Appending PATH environment variable: %s' % bin_path)
+        self.env_info.PATH.append(bin_path)
```
Comment (david-sinuela-pix4d-dev) [inline conanfile.py]: What does this do and how does it work?

Comment (krahenbuhl) [reply conanfile.py]: It comes from conanfile_installer.py, my understanding is that it adds sqlite3 to the path.

Comment (david-sinuela-pix4d-dev) [reply conanfile.py]: I would not add this unless we really understand what it does. It adds something to the path but when? when compiling the library? when consuming it? from cmake? how?

Comment (klaussnd) [reply conanfile.py]: Can we not pass the path to the executable when configuring `gdal`?

File: conanfile.py [hunk 3]
```diff
@@ -60,7 +61,7 @@ def _configure_cmake(self):
         cmake.definitions["HAVE_POSIX_FALLOCATE"] = True
         cmake.definitions["HAVE_STRERROR_R"] = True
         cmake.definitions["HAVE_USLEEP"] = True
-        cmake.definitions["BUILD_SHELL"] = False
+        cmake.definitions["BUILD_SHELL"] = True
```
Comment (klaussnd) [inline conanfile.py]: Should we make this optional?

File: conanfile.py (hunk 2 above)
Comment (krahenbuhl) [reply conanfile.py]: yes it's probably what should be done, what would be the best way to propagate the executable path via CMake? Presumably I need to modify the FindSQLite3.cmake but I would appreciate any guidance :)

File: conanfile.py (hunk 3 above)
Comment (krahenbuhl) [reply conanfile.py]: we could but I don't see under which circumstances it'd hurt to build it? if the need really arise we can always update the recipe

File: conanfile.py (hunk 2 above)
Comment (krahenbuhl) [reply conanfile.py]: I have now removed the setting of the path and now propagate the path the sqlite3 in the CMake finder instead, does that address your comments @david-sinuela-pix4d-dev @klaussnd  ?

Comment (david-sinuela-pix4d-dev) [reply conanfile.py]: Yes, much better!

**Pull request merged**